### DISCUSSION TITLE
Super Grammars

### DISCUSSION THREAD

**Discussion by cdietrich** [eclipse-langium/langium#1189] [General]
Hi, is there a reason the langium grammar seems to allow the declaration of super grammars but the feature does not seem to be implemented at all? are there any plans to support that.

- importing grammars works fine but this allows no tailoring / customization at the usage place 
- so the only option i currently see to achieve some level of customization is the usage of `import` in combination with `Guard Conditions`

**Comment by msujew** [comment]
The feature is only supported in the grammar to make it easier to migrate an Xtext based grammar to Langium. We're apparently just missing the validation to make that clear. The feature is not supported and currently not planned to be supported.

**Reply by cdietrich** [reply]
this also holds true for customizing imported grammars?

**Reply by msujew** [reply]
Yes, currently, all rules are final and cannot be customized (you should get an error trying to redefine an imported rule).
